src/config.rs

```diff
@@ -94,107 +178,188 @@ pub struct Style {
     pub bg: u16,
 }
 
-pub const CLEAR: Style =
-    Style {
-        fg: TB_DEFAULT,
-        bg: TB_DEFAULT,
-    };
-
-pub const USER_MSG: Style =
-    Style {
-        fg: 0,
-        bg: TB_DEFAULT,
-    };
-
-pub const ERR_MSG: Style =
-    Style {
-        fg: 0 | TB_BOLD,
-        bg: 1,
-    };
-
-pub const TOPIC: Style =
-    Style {
-        fg: 14 | TB_BOLD,
-        bg: TB_DEFAULT,
-    };
-
-pub const CURSOR: Style =
-    USER_MSG;
-
-pub const JOIN: Style =
-    Style {
-        fg: 10 | TB_BOLD,
-        bg: TB_DEFAULT,
-    };
-
-pub const PART: Style =
-    Style {
-        fg: 1 | TB_BOLD,
-        bg: TB_DEFAULT,
-    };
-
-pub const NICK: Style =
-    Style {
-        fg: 10 | TB_BOLD,
-        bg: TB_DEFAULT,
-    };
-
-pub const FADED: Style =
-    Style {
-        fg: 242,
-        bg: TB_DEFAULT,
-    };
-
-pub const EXIT_DIALOGUE: Style =
-    Style {
-        fg: TB_DEFAULT,
-        bg: 4,
-    };
-
-pub const HIGHLIGHT: Style =
-    Style {
-        fg: 9 | TB_BOLD,
-        bg: TB_DEFAULT,
-    };
-
-// Currently unused
-// pub const MENTION: Style =
-//     Style {
-//         fg: 220,
-//         bg: TB_DEFAULT,
-//     };
-
-pub const COMPLETION: Style =
-    Style {
-        fg: 84,
-        bg: TB_DEFAULT,
-    };
-
-pub const TIMESTAMP: Style = FADED;
-
-pub const TAB_ACTIVE: Style =
-    Style {
-        fg: 0 | TB_BOLD,
-        bg: 0,
-    };
-
-pub const TAB_NORMAL: Style =
-    Style {
-        fg: 8,
-        bg: 0,
-    };
-
-pub const TAB_NEW_MSG: Style =
-    Style {
-        fg: 5,
-        bg: 0,
-    };
-
-pub const TAB_HIGHLIGHT: Style =
-    Style {
-        fg: 9 | TB_BOLD,
-        bg: 0,
-    };
+const COLORS: [(&'static str, u16); 17] = 
+[
+    // Default bg color of the terminal
+    ("default",     TB_DEFAULT),
+
+    // Dark variants
+    ("black",       0),
+    ("darkred",     1),
+    ("darkgreen",   2),
+    ("darkyellow",  3),
+    ("darkblue",    4),
+    ("darkmagenta", 5),
+    ("darkcyan",    6),
+    ("lightgray",   7),
+    
+    // Bright variants
+    ("darkgray",    8),
+    ("red",         9),
+    ("green",       10),
+    ("yellow",      11),
+    ("blue",        12),
+    ("magenta",     13),
+    ("cyan",        14),
+    ("white",       15)
+];
+
+const ATTRS: [(&'static str, u16); 3] =
+[
+    ("bold",      TB_BOLD),
+    ("underline", TB_UNDERLINE),
+    ("reverse",   TB_REVERSE)
```

Comment:
I'm thinking maybe we should get rid of "reverse". I can't imagine anyone actually using this..